scripts/prepare_network.py

```diff
@@ -71,6 +74,46 @@
 logger = logging.getLogger(__name__)
 
 
+def download_emission_data():
+    try:
+        url = "https://jeodpp.jrc.ec.europa.eu/ftp/jrc-opendata/EDGAR/datasets/v60_GHG/CO2_excl_short-cycle_org_C/v60_GHG_CO2_excl_short-cycle_org_C_1970_2018.zip"
+        with requests.get(url) as rq:
+            with open("data/co2.zip", "wb") as file:
+                file.write(rq.content)
+        rootpath = os.getcwd()
+        file_path = os.path.join(rootpath, "data/co2.zip")
+        with ZipFile(file_path, "r") as zipObj:
+            zipObj.extract(
+                "v60_CO2_excl_short-cycle_org_C_1970_2018.xls", rootpath + "/data"
+            )
+        os.remove(file_path)
+        return "v60_CO2_excl_short-cycle_org_C_1970_2018.xls"
+    except:
+        logger.warning(f"Failed download resource from '{url}'.")
+        return False
+
+
+def emission_extractor(filename, emission_year):
+    country_names = n.buses.country.unique() 
+    # data reading process
+    datapath = os.path.join(os.getcwd(), "data", filename)
+    emission_of_countries = []
+    df = pd.read_excel(datapath, sheet_name="v6.0_EM_CO2_fossil_IPCC1996", skiprows=8)
+    df.columns = df.iloc[0]
+    df = df.set_index('Country_code_A3')
+    df = df.loc[df['IPCC_for_std_report_desc'] == "Public electricity and heat production"]
+    df = df.loc[:,'Y_1970':'Y_2018'].ffill(axis = 1)
+    df = df.loc[:,'Y_1970':'Y_2018'].bfill(axis = 1)
+    for j in country_names:
+        j = cc.convert(j, to="ISO3")
+        try:
+            emission_of_countries.append(df.loc[j]['Y_'+str(emission_year)])
+        except KeyError:
+            print(f"The emission value for {cc.convert(j, to='name_short', not_found=None)} is not found.")
+    emission_of_countries = sum(emission_of_countries)
+    return emission_of_countries
```

Comment:
This function is very interesting to get the list of emission by country.
I think it may be interesting to return a pandas series that specified the emission by country, then the sum may be performed afterwards.